src/core/provider/MCAccountValidator.cpp

```diff
@@ -0,0 +1,412 @@
+ For each service we test each server details (NetService),
+ Until either:  
+    we find on that works and returns ErrorNone in opCompleted().
+    we have gone trough the Array of NetService for that service and test() is recalled for the next service.
+ */
+void AccountValidator::test()
+{
+    if (mCurrentServiceTested == SERVICE_IMAP) {
+        if (mCurrentServiceIndex < mImapServices->count()) {
+            IMAPAsyncSession *imapSession = new IMAPAsyncSession();
+            imapSession->setUsername(mUsername);
+            imapSession->setPassword(mPassword);
+        
+            mImapServer = (NetService *) mImapServices->objectAtIndex(mCurrentServiceIndex);
+            imapSession->setHostname(mImapServer->hostname());
+            imapSession->setPort(mImapServer->port());
+            imapSession->setConnectionType(mImapServer->connectionType());
+            
+            mOperation = (IMAPOperation *)imapSession->checkAccountOperation();
+            mOperation->setCallback(this);
+            mOperation->start();
+        
+        }
+        else {
+            mCurrentServiceTested = SERVICE_POP;
+            mCurrentServiceIndex = 0;
+            test();
+        }
+    }
+    else if (mCurrentServiceTested == SERVICE_POP){
+        if (mCurrentServiceIndex < mPopServices->count()) {
+            POPAsyncSession *popSession = new POPAsyncSession();
+            popSession->setUsername(mUsername);
+            popSession->setPassword(mPassword);
+            
+            mPopServer = (NetService *) mPopServices->objectAtIndex(mCurrentServiceIndex);
+            popSession->setHostname(mPopServer->hostname());
+            popSession->setPort(mPopServer->port());
+            popSession->setConnectionType(mPopServer->connectionType());
+            
+            mOperation = (POPOperation *)popSession->checkAccountOperation();
+            mOperation->setCallback(this);
+            mOperation->start();
+        }
+        else {
+            mCurrentServiceTested = SERVICE_SMTP;
+            mCurrentServiceIndex = 0;
+            test();
+        }
+    }
+    else if (mCurrentServiceTested == SERVICE_SMTP){
+        if (mCurrentServiceIndex < mSmtpServices->count()) {
+            SMTPAsyncSession *smtpSession = new SMTPAsyncSession();
+            smtpSession->setUsername(mUsername);
+            smtpSession->setPassword(mPassword);
+            
+            mSmtpServer = (NetService *) mSmtpServices->objectAtIndex(mCurrentServiceIndex);
+            smtpSession->setHostname(mSmtpServer->hostname());
+            smtpSession->setPort(mSmtpServer->port());
+            smtpSession->setConnectionType(mSmtpServer->connectionType());
+            
+            mOperation =  (SMTPOperation *)smtpSession->checkAccountOperation(Address::addressWithMailbox(mEmail));
+            mOperation->setCallback(this);
+            mOperation->start();
+
+        }
+        else {
+           callback()->operationFinished(this);
```

Comment:
Which makes this code consistent with the other ones:
mCurrentServiceTested ++;
mCurrentServiceIndex = 0;
test();